.github/workflows/e2e-tests.yml

```diff
@@ -4,34 +4,39 @@ name: E2E Tests
 on: pull_request
 
 jobs:
+  setup-env:
+    name: 'Load ENV Vars'
+    uses: ./.github/workflows/setup-env.yml
+    secrets: inherit
+
   e2e-tests:
     runs-on: ubuntu-latest
+    needs: [setup-env]
+
     steps:
       - name: Checkout
         uses: actions/checkout@v4
       
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
         with:
-          node-version: "lts/*"
+          node-version: '20'
       - run: npm ci
 
       - name: 'Create env file'
         run: |
           touch .env
-          echo DEFAULTPROVIDER=${{ vars.DEFAULTPROVIDER }} >> .env
-          echo DEMO=${{ vars.DEMO }} >> .env
-          echo ENV=${{ vars.ENV }} >> .env
-          echo HOSTURL=${{ vars.HOSTURL }} >> .env
-          echo WEBHOOKHOSTURL=${{ vars.WEBHOOKHOSTURL }} >> .env
-          echo LOGLEVEL=${{ vars.LOGLEVEL }} >> .env
-          echo PORT=${{ vars.PORT }} >> .env
-          echo RESOURCEPREFIX=${{ vars.RESOURCEPREFIX }} >> .env
+          
+          # Vars
+          echo -e "${{ needs.setup-env.outputs.env_vars }}" >> .env
           echo RESOURCEVERSION="" >> .env
+
+          # Secrets (can't load these from another job, due to GH security features)
```

Comment:
I gave up on trying to share the env code after realizing how difficult it was to share secrets, but this looks like a great compromise. At least we can share the env variables